File: VB/WindowsFormsApplication154/CustomTextEdit.vb
Imports System
Imports System.ComponentModel
Imports DevExpress.XtraEditors


Namespace WindowsFormsApplication154

    <ToolboxItem(True)> _
    Public Class CustomTextEdit
        Inherits TextEdit

        Shared Sub New()
            RepositoryItemCustomTextEdit.RegisterCustomTextEdit()
        End Sub

        Public Sub New()
        End Sub

        <DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
        Public Shadows ReadOnly Property Properties() As RepositoryItemCustomTextEdit
            Get
                Return TryCast(MyBase.Properties, RepositoryItemCustomTextEdit)
            End Get
        End Property

        Public Overrides ReadOnly Property EditorTypeName() As String
            Get
                Return RepositoryItemCustomTextEdit.CustomEditName
            End Get
        End Property
    End Class
End Namespace

File: CS/WindowsFormsApplication154/CustomTextEdit.cs
using System;
using System.ComponentModel;
using DevExpress.XtraEditors;


namespace WindowsFormsApplication154 {

    [ToolboxItem(true)]
    public class CustomTextEdit : TextEdit {
        static CustomTextEdit() {
            RepositoryItemCustomTextEdit.RegisterCustomTextEdit();
        }

        public CustomTextEdit() {
        }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemCustomTextEdit Properties
        {
            get
            {
                return base.Properties as RepositoryItemCustomTextEdit;
            }
        }

        public override string EditorTypeName
        {
            get
            {
                return RepositoryItemCustomTextEdit.CustomEditName;
            }
        }
    }
}
